src/xform_to_datalog/datalog_facts.cc

```diff
@@ -0,0 +1,149 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+#include "absl/container/flat_hash_map.h"
+#include "src/xform_to_datalog/datalog_facts.h"
+
+#include "src/xform_to_datalog/arcs_manifest_tree/particle_spec.h"
+
+namespace raksha::xform_to_datalog {
+
+namespace ir = raksha::ir;
+namespace types = raksha::ir::types;
+namespace amt = raksha::xform_to_datalog::arcs_manifest_tree;
```

Comment:
`amt` -> `arcs`

Reply:
The issue with doing that is that it causes a name collision: the Arcs manifest proto library already introduces an `arcs` namespace, in which proto classes such as `ParticleSpecProto` live.